### PR TITLE
Support for Virt driver

### PR DESCRIPTION
**Pull request by afbjorklund**
This is a driver based on [libvirt.org,](https://libvirt.org) for remote hypervisors.

The libvirt functions are in a plugin, to minimize dependencies.

----

Currently only shows version:

```go
var VERSION uint32 = libvirt.VERSION_NUMBER

func Version() (uint32, error) { return libvirt.GetVersion() }
```

To prove that it can access the API.

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -30,3 +30,9 @@ config GUESTAGENT_ARCH_RISCV64
 	help
 		Build lima-guestagent for "riscv64" Arch
 	default y
+
+config PLUGIN_VIRT
+	bool "plugin for vmType virt"
```
Comment (AkihiroSuda) [inline Kconfig:35]: I think this should be called "libvirt"

Comment (AkihiroSuda) [reply Kconfig:35]: I also wonder if we can split out this to a separate plugin binary.

Same applies to the vbox driver, cloud drivers, etc.

Comment (afbjorklund) [reply Kconfig:35]: Sure, it just requires more work. And it would be needed, for cloud.
* https://github.com/lima-vm/lima/discussions/2007

**Comment by afbjorklund** [comment]
Apparently this experiment does not work, since putting libvirt in go.mod requires libvirt-dev:

```
 No package 'libvirt' found
  Package libvirt was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libvirt.pc'
  to the PKG_CONFIG_PATH environment variable
```

**Comment by afbjorklund** [comment]
1) Running "session" qemu with libvirt doesn't add any except complexity (until there is proper "system" qemu support)
2) Trying to load libvirt from a plugin doesn't work due to go modules, needs a separate binary like docker-machine-kvm

**Comment by afbjorklund** [comment]
All that you needed to do was to set the build tag `libvirt_dlopen`, then it doesn't need pkgconfig or the libraries...

**Comment by afbjorklund** [comment]
Hmm, libvirt has dozens of connection points (function wrappers) so maybe it should be a plugin after all.

**Comment by AkihiroSuda** [comment]
Would it be possible to just exec `virsh`?

**Comment by afbjorklund** [comment]
Yes, that should work. Something like `virsh create {{.Dir}}/domain.xml`

**Pull request closed without merging**